mail_debrand/models/ir_qweb.py

```diff
@@ -0,0 +1,16 @@
+from odoo.addons.base.models.qweb import QWeb
+from odoo import models
+
+
+class IrQWeb(models.AbstractModel, QWeb):
+    _inherit = 'ir.qweb'
+
+    def render(self, template, values=None, **options):
+        result = super(IrQWeb, self).render(template, values=values, **options)
+        lang = values.get("lang", False)
+        message = values.get("message", False)
+        if lang and message:
```

Comment:
> Why are you checking for `message` if then you always debrand all the `result`?
> 
> Moreover, `values` is an arbitrary dictionary passed to a mail template so we are not sure that language's variable is named `lang`; is it possible to get directly the language used for rendering?
> From https://github.com/odoo/odoo/blob/2be47637dd5cc48ca561ff85cf2d359dde0829ab/odoo/addons/base/models/qweb.py#L1506-L1554 looks like the language will be in `self`'s context.
> 

I have to be sure if `render` method of IrQWeb is called after `_notify`, just before an email sending.
From [https://github.com/odoo/odoo/blob/12.0/addons/mail/models/res_partner.py#L37-L86](https://github.com/odoo/odoo/blob/12.0/addons/mail/models/res_partner.py#L37-L86) you have both `message` and `lang`, and that's when you want to debrand (`lang` can be explicitly set in `mail.template`).

> Can you add a test for the use-case you have fixed?

WIP ;)

Reply:
> Checking for lang and message can be good for the specific case when a message is sent to a partner because the method you have linked _notify_prepare_template_context is called just before render in _notify: https://github.com/odoo/odoo/blob/5d40023eaf9b3c25941429e7fac266722780846d/addons/mail/models/res_partner.py#L107-L136
> 
> The problem is that render can be called in other situations too where there is no lang or message in values, the debranding should happen in those cases too.

This module debrands only `mail.message` html body (that includes mails, internal messages and notes), and this is why I wrote the condition. When `_notify_prepare_template_context` is called, I'm sure that in `render`'s `value` are stored a `mail.message` and `lang`. Why should it include other cases? There are other situations where a `mail.message` is created outside `_notify`?